File: java/src/test/java/com/alxarafe/domain/topology/entity/ZoneTypeTest.java
package com.alxarafe.domain.topology.entity;

import com.alxarafe.app.domain.topology.entity.ZoneType;
import com.alxarafe.app.domain.topology.valueobject.ZoneTypeCode;
import com.alxarafe.app.domain.topology.valueobject.ZoneTypeId;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ZoneTypeTest {

    @Test
    void create() {
        var id = new ZoneTypeId("018e4e3a-3e7b-7b3e-8000-000000000001");
        var code = new ZoneTypeCode("PICKING");
        var zoneType = new ZoneType(id, code, true, false);

        assertEquals(id, zoneType.id());
        assertEquals(code, zoneType.code());
        assertTrue(zoneType.isOperative());
        assertFalse(zoneType.allowsMultiSku());
    }
}

File: php/tests/Unit/Domain/Topology/Entity/ZoneTypeTest.php
<?php
declare(strict_types=1);

namespace Tests\Unit\Domain\Topology\Entity;

use Alxarafe\App\Domain\Topology\Entity\ZoneType;
use Alxarafe\App\Domain\Topology\ValueObject\ZoneTypeCode;
use Alxarafe\App\Domain\Topology\ValueObject\ZoneTypeId;
use PHPUnit\Framework\TestCase;

final class ZoneTypeTest extends TestCase
{
    public function testCreate(): void
    {
        $id = new ZoneTypeId('018e4e3a-3e7b-7b3e-8000-000000000001');
        $code = new ZoneTypeCode('PICKING');
        $zoneType = new ZoneType($id, $code, true, false);

        self::assertTrue($id->equals($zoneType->id()));
        self::assertTrue($code->equals($zoneType->code()));
        self::assertTrue($zoneType->isOperative());
        self::assertFalse($zoneType->allowsMultiSku());
    }
}
